contracts-periphery/src/WithdrawHook.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
```

Comment:
Is there an issue somewhere describing the spec/scope of this PR? If so we should link that in the PR description. I think I get the gist of it, but am unable to verify that this PR actually implements what we want.

Example questions:
- The Uniswap router lets you swap with V2 and V3, but this hook only supports Uniswap V3.
- I believe the Uniswap router lets the integrator charge their own fee (and if it doesn't, that's something we could add ourselves). Having the option to charge our own fee seems useful.
- This only supports one swap, so a user can't receive DAI then swap some to ETH, some to USDC, and send both to the same wallet.

Reply:
1. To support V2, would it be better to pass in a user param that specifies they're using V2 and add additional functions that do the swaps using V2's `_swap(address[] memory path, address _to)` OR should the only function just call `swapRouter.call(data)` data being the one from Auto Router API.
2. Implemented `unwrapWETH9WithFee`
3. One could argue that the user can just swap some DAI to ETH, then once withdrawn to a new address, they can swap ETH to USDC, which would keep the contract much simpler. 

But if we were to accommodate that scenario, I can think of an implementation that only accommodates that particular use case of swapping some for ETH and some for another token. If we want the contract to be more flexible and maybe support multiple swaps, i.e. some for eth, some for different token1 and token2, etc., I could try to implement that using multiple loops, please let me know if there's a simpler and more gas efficient way to do things that I might be missing.

Reply:
I see that user passing in the multicall data might be the easiest way as you mentioned in https://github.com/ScopeLift/umbra-protocol/pull/341#discussion_r890298937 . I was just struggling to balance how much of the logic should be done within the contract vs. by the front end that's calling it

Reply:
Ok you and @apbendi agree on not supporting two token swaps and that we should only support a single swap to ETH. So my suggestion is that we just have the swap method take in the encoded multicall data that gets called on the `SwapRouter02` as mentioned in https://github.com/ScopeLift/umbra-protocol/pull/341#discussion_r890298937. This seems to solve all our problems:
- Supports Uni V2, V3, or a combination
- Lets use us their fee methods instead of re-implementing our own
- Our contract becomes very simple and just needs a single method, doesn't even need to decode the data, and just pass it on to the uniswap contract. Maybe we decode to verify the output token or something, but doesn't really seem necessary since the contract is basically just a pass-through to the router which handles the checks. 

Is there a reason not to handle it this way?

Reply:
> I was just struggling to balance how much of the logic should be done within the contract vs. by the front end that's calling it

The answer definitely varies depending on what you're doing specifically, but I don't see any benefit to doing this on-chain for our use case. It will just be more expensive, less flexible, and Uniswap's router has the required checks we need anyway

Reply:
We're now taking the simpler approach in https://github.com/ScopeLift/umbra-protocol/pull/341/commits/b40d4a8da174dc150d24e62c76a20189c2d8cfc2